src/dataServices/hybridDataService.ts

```diff
@@ -251,14 +251,10 @@ export class HybridDataService implements Persistence, BulkDataAccess {
         const succeedDbItemResults = deletionDbItemResults.filter((x) => !(x instanceof Error));
         const failedDbItemResults = deletionDbItemResults.filter((x) => x instanceof Error);
 
-        const deleteBulkPromsises: Array<Promise<any>> = [];
-        succeedDbItemResults.forEach((delBulk: any) => {
-            if (delBulk.bulkDataLink) {
-                deleteBulkPromsises.push(S3ObjectStorageService.deleteObject(delBulk.bulkDataLink));
-            }
-        });
-
-        await Promise.all(deleteBulkPromsises);
+        const bulkDelPromises = succeedDbItemResults
+            .filter((currentResult) => currentResult.bulkDataLink)
```

Comment:
`filter` doesn't change the type of the elements, so in subsequent `map` I would expect the same temp name as in the filter.